pilot/pkg/networking/core/v1alpha3/envoyfilter/listener_patch.go

```diff
@@ -305,6 +305,13 @@ func doNetworkFilterOperation(patchContext networking.EnvoyFilter_PatchContext,
 			var retVal *any.Any
 			if userFilter.GetTypedConfig() != nil {
 				// user has any typed struct
+				// The type may not match up exactly. For example, if we use v2 internally but they use v3.
+				// Assuming they are not using deprecated/new fields, we can safely swap out the TypeUrl
+				// If we did not do this, proto.Merge below will panic (which is recovered), so even though this
+				// is not 100% reliable its better than doing nothing
+				if userFilter.GetTypedConfig().TypeUrl != filter.GetTypedConfig().TypeUrl {
```

Comment:
This will work. It means that you do not need to specify typeURL in the patch.
1. It only addresses the merge path though, Will v2 / v3 mixed additional continue to work?
Otherwise we can upconvert all v2 typeURLs to v3 typeurl.
2. If we erroneously set the type URL, what is the error message in the panic? 
A clear message like `cannot change type URL from `type.googleapis.com/envoy.config.filter.http.fault.v2.HTTPFault` to `type.googleapis.com/envoy.config.filter.http.fault.v3.echo`

Reply:
Yes additions should work, Envoy can properly handle mixed versions. The only issue was merging.